src/keygen/keyshare.rs

```diff
@@ -51,9 +65,29 @@ impl KeySharePublic {
         }
     }
 
+    /// Get the public curvepoint which is the public key share of the
+    /// participant.
+    pub fn X(&self) -> CurvePoint {
```

Comment:
suggestion: I don't know when we're going to change the name of the field but typically capital letters are reserved for types in Rust. The library currently suppresses naming errors because there are so many, but I might rename this to, like `as_point` to be more standard.

Reply:
I just learned from @picklenerd about using `AsRef`. The theory is, instead of having this function we do 
```rust
impl AsRef<CurvePoint> for KeySharePublic
```

and then replace all calls to `public_share.X()` with plain `public_share.as_ref()`. This is nicer because it's (a) canonical Rust and (b) avoids some implication that public share has other stuff going on in it. 

I think that would be a great improvement but I don't want to get completely carried away because I think this pattern could be applied on other types as well. Maybe we could even look into the places where we call `as_ref` and see if we can rewrite those call spots to take `AsRef<CurvePoint>` as a parameter instead of straight `&CurvePoint`, so we don't have to manually call the function at all. I think this would make bits of the protocol code more readable. But that's way too much to do in this PR.

So: action items
- Hridam will change this function. You can either to `as_point` or implement `AsRef<CurvePoint>`.
- I will write an issue outlining other concrete spots where we could use `AsRef` for improved readability.